library/rsa.c

```diff
@@ -1429,15 +1442,27 @@ int mbedtls_rsa_rsassa_pkcs1_v15_verify( mbedtls_rsa_context *ctx,
     if( asn1_len != hashlen )
         return( MBEDTLS_ERR_RSA_VERIFY_FAILED );
 
-    if( memcmp( p, hash, hashlen ) != 0 )
-        return( MBEDTLS_ERR_RSA_VERIFY_FAILED );
+    ret = MBEDTLS_ERR_RSA_VERIFY_FAILED;
+    ret = memcmp( p, hash, hashlen );
 
-    p += hashlen;
+    if( ret == 0 )
+    {
+        if( ret != 0)
```

Comment:
This line is logically impossible isn't it? To reach here `ret == 0`, and then you test that `ret != 0`.